src/Sylius/Behat/Context/Api/Shop/LoginContext.php

```diff
@@ -262,17 +262,30 @@ public function iShouldSeeWhoIAm(): void
         );
     }
 
+    /**
+     * @Then I should be notified that the user email does not exist
+     */
+    public function iShouldBeNotifiedThatTheUserDoesNotExist(): void
+    {
+        Assert::same($this->client->getLastResponse()->getStatusCode(), 422);
+    }
+
```

Comment:
I'm uncertain whether we should depend only on the status code. The reason behind might be any, and we should check against the message too.

Reply:
Checking the message is better indeed, it will help keeping the behavior of the API.

Reply:
A specific message could be asserted in a test, but it wouldn't be very reliable if downstream code ended up working that way too, because 1. messages may change over time and 2. they can be localized. I'm not sure it's a good idea to assert more in a test than what can be relied on in actual code.

Reply:
In most places we put a raw string in English with an expected message (all tests go with en_US, or at least most of them).

If you like, I saw once a test where a translator service was injected, and used to validate the message.

> 1. messages may change over time
A person changing the message may fix the test. We have a ton of asserts where we put a raw string to validate the message, and never had any bigger troubles with that.

> 2. they can be localized
It's E2E test, we know at what exactly locale we are, so it isn't a problem.

Reply:
For reference, use the "Find in files" feature in your IDE and paste "iShouldBeNotified".